bin/resource_management/url_database.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import grequests
+import fileinput
+from datetime import datetime
+import json
+import csv
+
+"""
+Fetches URL's from metadata and tests URL status in parallel using grequests library.
+Outputs status to 'url_database.csv'
+
+Arguments:
+fileinput - metadata file to be read ('od-do-canada.jsonl')
+batch_size - maximum number of URL's to test in parallel
+"""
+
+batch_size = int(sys.argv[2])
+
+i = 1
+prev_i = 1
+urls = []
+batch_urls = []
+responses = []
+date = []
+
+
+print("Starting...")
+print("Reading and testing URL's")
+
+for dataset in fileinput.input():
+    line = json.loads(dataset)
+    resources = line["resources"]
+    for l in range(len(resources)):
+        url = resources[l]["url"]
+        if url in urls:
+            continue;
+        else:
+            urls.append(url)
+            now = datetime.now()
+            dt_string = now.strftime("%d/%m/%Y %H:%M")
+            date.append(dt_string)
+            batch_urls.append(url)
+            if len(batch_urls)==batch_size:
+                sys.stderr.write("\r")
+                sys.stderr.write("Testing Datasets {0} - {1}".format(prev_i, i))
+                prev_i=i

```

Comment:
pep8: missing spaces around `=` when used as an assignment (no spaces when used for keyword args)